cmd/dcrtime_checker/dcrtime_checker.go

```diff
@@ -0,0 +1,227 @@
+package main
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/decred/dcrd/txscript"
+	"github.com/decred/dcrdata/v4/api/types"
+	v1 "github.com/decred/dcrtime/api/v1"
+	"github.com/decred/dcrtime/merkle"
+)
+
+var (
+	proof       = flag.String("p", "", "Proof file")
+	file        = flag.String("f", "", "Original file")
+	dcrdataHost = flag.String("h", "", "dcrdata host")
+	testnet     = flag.Bool("testnet", false, "Use testnet port")
+	verbose     = flag.Bool("v", false, "Verbose")
+)
+
+// normalizeAddress returns addr with the passed default port appended if
+// there is not already a port specified.
+func normalizeAddress(addr, defaultPort string) string {
+	_, _, err := net.SplitHostPort(addr)
+	if err != nil {
+		return net.JoinHostPort(addr, defaultPort)
+	}
+	return addr
+}
+
+// digestFile returns the SHA256 of a file.
+func digestFile(filename string) (string, error) {
+	h := sha256.New()
+	f, err := os.Open(filename)
+	if err != nil {
+		return "", err
+	}
+	defer f.Close()
+	if _, err = io.Copy(h, f); err != nil {
+		return "", err
+	}
+
+	return hex.EncodeToString(h.Sum(nil)), nil
+}
+
```

Comment:
would be nice to have a quick explanation of what this function is doing

Reply:
These functions really need to move into util. Will do so.